servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/AbstractHttpServiceAsyncContextTest.java

```diff
@@ -601,12 +691,24 @@ public void onRequestError(Throwable cause) {
 
         @Override
         public void onRequestCancel() {
-            errorQueue.add(new AssertionError("Unexpected onRequestCancel"));
+            switch (responseType) {
+                case ERROR_ON_RESPONSE:
+                case ERROR_ON_RESPONSE_BODY:
+                case CANCEL_ON_RESPONSE:
+                    assertAsyncContext();
+                    break;
+                default:
+                    errorQueue.add(new AssertionError("Unexpected onRequestCancel"));
```

Comment:
WDYT if we relax these assertions (here and in all other callbacks) to only call `assertAsyncContext()` and never check for `Unexpected <method>`? Cancelation is racy and it's hard to make it consistent. In these tests we are not testing what callbacks are reported, we only try to make sure the context is propagated consistently to all callbacks.
I think those were originally added because it was known those callbacks should not be invoked (we only had success cases) just to not leave them empty.
Alternatively, you can keep `Unexpected <method>` only for `responseType != SUCCESS`.

Reply:
I think I had them sorted out, but ultimately I don't think they added much extra value so I took your second suggestion and only add errors when we get failures when we expect success.